### ISSUE TITLE
`phpdoc_types_order` bug with intersection types and generic annotations

### ISSUE THREAD

**Issue by AlexandruGG** [FriendsOfPHP/PHP-CS-Fixer#5885]
## Bug report

3.0.1 (or 3.0.2) seems to have introduced a bug with `phpdoc_types_order`.
I found a related issue https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/5798 which was closed as solved. 
I think this is a slightly different issue but still related.

### Code snippet that reproduces the problem

**Initial code:**
```PHP
/**
 * @var Collection<int, AbstractInvoice> & Selectable<int, AbstractInvoice>
 */
private Collection $invoices;
```

**Changed code:**
```PHP
/**
 * @var Collection<int, <|AbstractInvoice> & Selectable|int, AbstractInvoice>
 */
private Collection $invoices;
```

_Current workaround I'm using:_
```PHP
/**
 * @var (Collection<int, AbstractInvoice>) & (Selectable<int, AbstractInvoice>)
 */
private Collection $invoices;
```

**Comment by andyexeter** [comment]
I believe this has been fixed in #6063

**Comment by SpacePossum** [comment]
Thanks for the headsup, this seems fixed indeed in v3.6.0, let me know if not and I'll reopen :+1:
